.github/workflows/build.yml

```diff
@@ -0,0 +1,81 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install Zeo++
+      run: |
+        git clone https://github.com/pzarabadip/zeoplusplus -b develop zeoplusplus
+        cd zeoplusplus
+        cd voro++/src
+        make
+        cd ../..
+        make
+        mkdir bin
+        cp network bin
+    - name: Upload network executable
```

Comment:
Hmm... is this is actually uploading the artifact every time?

Actually, looking into github action's documentation, it seems they *do* provide a caching mechanism and they suggest to use it for our case:
https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#comparing-artifacts-and-dependency-caching

Is there a reason why you decided against this approach?

Reply:
Currently, using the cache comes with some issues:
https://github.com/actions/cache/issues/144

I have tried it at the beginning and was facing issues too. For now, I made it through artifacts until this issues is resolved.